src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3549,6 +3550,7 @@ void MergeTreeData::changeSettings(
         auto copy = getDefaultSettings();
         copy->applyChanges(new_changes);
         copy->sanityCheck(getContext()->getMergeMutateExecutor()->getMaxTasksCount());
+        checkColumnFilenamesForCollision(*copy, /*throw_on_error=*/ true);
```

Comment:
Should we also check it when altering columns?

Reply:
Yes, makes sense.